### ISSUE TITLE
Orphaned host/service checks needs to be rescheduled immediately

### ISSUE THREAD

**Issue by ranjithkodumbu** [NagiosEnterprises/nagioscore#675]
Hi Team,

I have noticed that orphaned host/service checks are not rescheduled immediately, due to this there is a high chance of missing the alerts. Please have a look into this.

Regards,
Ranjith Kumar R

**Comment by Madlohe** [comment]
Hi @ranjithkodumbu,

In nagios.cfg, what are the values of `check_for_orphaned_services` and `check_for_orphaned_hosts`? If these are both set to `1`, nagios should always immediately reschedule an orphaned check.

**Comment by ranjithkodumbu** [comment]
Yes, check_for_orphaned_services And check_for_orphaned_hosts Set to 1 in nagios.cfg. However I request you to validate the code.
Thank you.

**Comment by Madlohe** [comment]
As far as I can tell, that check scheduling code should always execute (so long as the expected check time is before the current time). The relevant code is at https://github.com/NagiosEnterprises/nagioscore/blob/1ebaddc9d469b0d5d18f7557ad86b1b10d1ed26a/base/checks.c#L1999-L2048 
and 
https://github.com/NagiosEnterprises/nagioscore/blob/1ebaddc9d469b0d5d18f7557ad86b1b10d1ed26a/base/checks.c#L2745-L2797
Each of these functions will run every minute. You can verify that they're running on your own system by setting `debug_level=1` in nagios.cfg. You can then watch the debug log for these messages with 
`tail -f /usr/local/nagios/var/nagios.debug | grep orphan`.

What are you using to determine that checks were orphaned and not rescheduled?

**Comment by ranjithkodumbu** [comment]
Thanks a lot for the information, I will monitor the debug log and let you know if I find something wrong. I appreciate your efforts put in to identify the code.

**Comment by Madlohe** [comment]
Hi @ranjithkodumbu,

Any update on this? The problem described is relatively serious, so I'd like to get it fixed if it truly is an issue.

**Comment by ranjithkodumbu** [comment]
Hi Madlohe,

I just checked, looks like " 'CpuUtilization'  service didn't rescheduled. Though the log says  "I'm scheduling an immediate check of the service...". I never rescheduled the check. I think there might be problem with scheduler.

[1566232695] Warning: The check of service 'CpuUtilization' on host 'test.com' looks like it was orphaned (results never came back; last_check=1566230779; next_check=1566231835).  I'm scheduling an immediate check of the service...

Regards,
Ranjith Kumar R
